components/contract-functions/contract-uri-upload.tsx

```diff
@@ -0,0 +1,48 @@
+import { Icon, Input, InputGroup, InputRightElement } from "@chakra-ui/react";
+import { useStorageUpload } from "@thirdweb-dev/react";
+import { FileInput } from "components/shared/FileInput";
+import { FiUpload } from "react-icons/fi";
+import { Button } from "tw-components";
+
+interface ContractUriUploadProps {
+  value: string;
+  setValue: (value: string) => void;
+}
+
+export const ContractUriUpload: React.FC<ContractUriUploadProps> = ({
+  value,
+  setValue,
+}) => {
+  const { mutateAsync: upload, isLoading } = useStorageUpload();
```

Comment:
if u use mutateAsync that means you *have* to `try{}catch{}`it (it will throw if it fails)